### PR TITLE
pr update sn api 0.18.0

### PR DESCRIPTION
**Pull request by b-zee**
- feat!: update neon and cli dep
- feat: fixing argument and function name changes

### CODE REVIEW

**Comment by b-zee** [comment]
@joshuef @bochaco Is there a workflow from another project that I might use as an example of how to make the tests work properly?

**Comment by bochaco** [comment]
All tests call `new_safe` helper function to instantiate the Safe API instance using a hard-coded string as the credentials: https://github.com/maidsafe/sn_nodejs/blob/master/test/helpers.js#L6. And they now all would need to run against a real local network, so you'd need that first, run a local network.

That API changed recently and you need to pass a keypair as credentials to `connect` (https://github.com/maidsafe/sn_api/blob/master/sn_api/src/api/app/auth.rs#L74), you could generate one with CLI by self authorising and then deserialising the content of `~/.safe/cli/credentials`. Or perhaps you can generate a keypair with CLI (`$ safe keys create --test-coins`) or sn_api possibly (https://github.com/maidsafe/sn_api/blob/master/sn_api/src/api/app/auth.rs#L74). Once you get the `Safe` instance created correctly the tests should be able to connect to the local network and operate on it.

I'm mentioning manual steps here for you to play with and to see if it works, then the challenge is to automate that so it's part of the CI/testing scripts of this crate. I hope it helps.

**Comment by b-zee** [comment]
Thanks for the helpful pointers, @bochaco!

While I was implementing some of the functionality I came across two challenges.

First, I implemented Keypair (passed to `Safe::connect`) and SecretKey (passed to `keys_balance_from_sk`). Just to see if this worked and was possible with Neon. The problem here is that SecretKey must be **moved** when calling `keys_balance_from_sk`. But, if a JsSecretKey object is passed to a function, we can only **borrow** the internal SecretKey. I can't yet wrap my head around how I should exactly solve this with the current API.

Second, Neon doesn't seem to allow one to wrap a struct instance into a class instance without calling the constructor. But, in the case of SecretKey, a constructor doesn't make sense. It's never instantiated from JavaScript or within the binding, they are returned from the Safe APIs. (With the native N-API this should be possible. I might look into [napi-rs](https://napi.rs/recipes/define-class), what do you think @joshuef?)

I've solved these by constructing a SecretKey from a Keypair, and requiring a Keypair in `keys_balance_from_sk`. In this way, the Keypair that is passed can produce an owned SecretKey that is consumed by the Rust function.

<hr>

Now, this is the first little hurdle and I feel there are possibly many to come. I'm learning a lot, but it requires a lot of effort (obviously) to understand the rationale behind the Rust API combined with creating the right bindings.
I am also a little anxious about the fact that most calls are implemented as blocking, while I've learned that Node.js is mostly single-threaded.

**Comment by joshuef** [comment]
Interesting. I haven't heard of the n-api rust lib (bin a while since we set up neon). but from reading around (https://napi.rs/neon and https://www.reddit.com/r/rust/comments/kmwkla/napi100_released_build_fast_and_safe_nodejs/ghi9nwe/) make it seem like n-api may well be a good move.

We're certainly not wedded to neon (the lack of full async has been something we've just had to accept for the time being). So if you fancy trying out n-api, even a small poc would be great to see!

(havent scanned the PR at mo, will swing back around to this later the day hopefully 👍 )

File: .github/workflows/pr.yml [hunk 1]
```diff
@@ -114,12 +114,6 @@ jobs:
       #     $HOME/.safe/node/sn_node.exe -V
       #     safe --version
 
-      - run : yarn
```
Comment (S-Coyle) [inline .github/workflows/pr.yml:117]: I think we still need this line for the steps below which include `yarn ...`

**Comment by b-zee** [comment]
Closing for now, in favor of the `napi-rs` version of this library.

**Pull request closed without merging**